src/legacy/core_plugins/interpreter/public/functions/esaggs.ts

```diff
@@ -32,37 +36,56 @@ import chrome from 'ui/chrome';
 const courierRequestHandlerProvider = CourierRequestHandlerProvider;
 const courierRequestHandler = courierRequestHandlerProvider().handler;
 
-export const esaggs = () => ({
-  name: 'esaggs',
+import { ExpressionFunction } from '../../types';
+import { KibanaContext, KibanaDatatable } from '../../common/types';
+
+const name = 'esaggs';
+
+type Context = KibanaContext | null;
+
+interface Arguments {
+  index: string | null;
+  metricsAtAllLevels: boolean;
+  partialRows: boolean;
+  aggConfigs: string;
+}
+
+type Return = Promise<KibanaDatatable>;
+
+export const esaggs = (): ExpressionFunction<typeof name, Context, Arguments, Return> => ({
+  name,
   type: 'kibana_datatable',
   context: {
-    types: [
-      'kibana_context',
-      'null',
-    ],
+    types: ['kibana_context', 'null'],
   },
-  help: i18n.translate('interpreter.functions.esaggs.help', { defaultMessage: 'Run AggConfig aggregation' }),
+  help: i18n.translate('interpreter.functions.esaggs.help', {
```

Comment:
You might want to take a look at how we're enforcing `i18n` in Canvas... because if you use the same methodology, we can move those types to OSS as well, (in a future PR).

https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/canvas_plugin_src/strings/functions/function_help.ts#L79...L97

Reply:
Thanks! Will check it out